### PR TITLE
Improve SqlDataRecord test coverage

### LINKED ISSUE #27416: Improve SqlDataRecord test coverage
Improve SqlDataRecord test coverage. 
cc @remibou

### PR DESCRIPTION
**Pull request by RemiBou**
Improved code coverage (+83 lines), added some tests for edge case, see #27416 

Fixes #27416
cc @saurabh500

### CODE REVIEW

**Comment by ahsonkhan** [comment]
@dotnet-bot test Windows x86 Release Build

**Comment by ahsonkhan** [comment]
Ubuntu.1404.Amd64.Open:Release-x64
System.Security.Cryptography.Xml.Tests
https://mc.dot.net/#/user/RemiBou/pr~2Fjenkins~2Fdotnet~2Fcorefx~2Fmaster~2F/test~2Ffunctional~2Fcli~2F/d7680bb69055ec516af57860125fc44b89576e31/workItem/System.Security.Cryptography.Xml.Tests/wilogs

```text
2018-02-24 00:05:12,942: INFO: proc(54): run_and_log_output: Output: Starting:    System.Security.Cryptography.Xml.Tests
2018-02-24 00:05:15,318: INFO: proc(54): run_and_log_output: Output: /home/helixbot/dotnetbuild/work/4020940d-0185-4380-b734-80f239965da4/Work/06f29225-a1a4-4622-93e9-8dcd849e77a5/Unzip/RunTests.sh: line 88: 20449 Aborted                 (core dumped) $RUNTIME_PATH/dotnet xunit.console.netcore.exe System.Security.Cryptography.Xml.Tests.dll -xml testResults.xml -notrait Benchmark=true -notrait category=nonnetcoreapptests -notrait category=nonlinuxtests -notrait category=OuterLoop -notrait category=failing
2018-02-24 00:05:15,360: INFO: proc(54): run_and_log_output: Output: Dumpling has not been installed yet. Please run "DumplingHelper.py install_dumpling" before collect_dumps.
2018-02-24 00:05:15,363: INFO: proc(54): run_and_log_output: Output: ~/dotnetbuild/work/4020940d-0185-4380-b734-80f239965da4/Work/06f29225-a1a4-4622-93e9-8dcd849e77a5/Unzip
2018-02-24 00:05:15,364: INFO: proc(54): run_and_log_output: Output: ----- end 00:05:15 ----- exit code 134 ----------------------------------------------------------
2018-02-24 00:05:15,366: INFO: proc(54): run_and_log_output: Output: Unable to find executable corerun
2018-02-24 00:05:15,369: INFO: proc(54): run_and_log_output: Output: Copying core file core to /tmp/coredumps
2018-02-24 00:05:15,576: INFO: proc(58): run_and_log_output: Exit Code: 134
2018-02-24 00:05:15,577: ERROR: scriptrunner(91): _main: Error: No exception thrown, but XUnit results not created
2018-02-24 00:05:15,577: ERROR: helix_test_execution(83): report_error: Error running xunit None
```

cc @bartonjs, @dnceng

@dotnet-bot test Linux x64 Release Build

File: src/System.Data.SqlClient/tests/FunctionalTests/SqlDataRecordTest.cs [hunk 1]
```diff
@@ -136,6 +138,260 @@ public void SqlRecordFillTest()
                 Assert.Equal(DBNull.Value, record.GetValue(i));
             }
         }
+        [Fact]
+        public void GetDataTypeName_ReturnsMetaDataTypeIfUdtType()
+        {
+            SqlMetaData[] metaData = new SqlMetaData[]
+            {
+                new SqlMetaData("col1", SqlDbType.Udt,typeof(TestUdt),"sql_TestUdt")
```
Comment (saurabh500) [inline src/System.Data.SqlClient/tests/FunctionalTests/SqlDataRecordTest.cs]: Nit: Space after `, "sql_TestUdt"`

File: src/System.Data.SqlClient/tests/FunctionalTests/SqlDataRecordTest.cs [hunk 2]
```diff
@@ -136,6 +138,260 @@ public void SqlRecordFillTest()
...
+
+            Assert.Equal("System.Data.SqlClient.Tests.TestUdt", record.GetDataTypeName(0));
+        }
+
+        [Fact]
+        public void GetDataTypeName_ReturnsTypeFromMetaTypeIfNotUdt()
+        {
+            SqlMetaData[] metaData = new SqlMetaData[]
+            {
+                new SqlMetaData("col1", SqlDbType.NVarChar,50)
```
Comment (saurabh500) [inline src/System.Data.SqlClient/tests/FunctionalTests/SqlDataRecordTest.cs]: Nit: Space after `,` in `, 50`

File: src/System.Data.SqlClient/tests/FunctionalTests/SqlDataRecordTest.cs [hunk 3]
```diff
@@ -136,6 +138,260 @@ public void SqlRecordFillTest()
...
+
+            Assert.Equal(typeof(string), record.GetFieldType(0));
+        }
+
+        [Fact]
+        public void GetValues_ThrowsIfNull()
+        {
+            SqlMetaData[] metaData = new SqlMetaData[]
+            {
+                new SqlMetaData("col1", SqlDbType.NVarChar,50)
```
Comment (saurabh500) [inline src/System.Data.SqlClient/tests/FunctionalTests/SqlDataRecordTest.cs]: Nit: Space after `,` in `, 50`

File: src/System.Data.SqlClient/tests/FunctionalTests/SqlDataRecordTest.cs [hunk 4]
```diff
@@ -136,6 +138,260 @@ public void SqlRecordFillTest()
...
+            SqlDataRecord record = new SqlDataRecord(metaData);
+
+            Assert.Equal("nvarchar", record.GetDataTypeName(0));
+        }
+        [Fact]
+        public void GetFieldType_ReturnMetaTypeClassType()
+        {
+            SqlMetaData[] metaData = new SqlMetaData[]
+            {
+                new SqlMetaData("col1", SqlDbType.NVarChar,50)
```
Comment (saurabh500) [inline src/System.Data.SqlClient/tests/FunctionalTests/SqlDataRecordTest.cs]: Nit: Space after `,` in `, 50`

File: src/System.Data.SqlClient/tests/FunctionalTests/SqlDataRecordTest.cs [hunk 5]
```diff
@@ -136,6 +138,260 @@ public void SqlRecordFillTest()
...
+
+            Assert.Throws<ArgumentNullException>(() => record.GetValues(null));
+        }
+
+        [Fact]
+        public void GetValues_IfValuesBiggerThanColumnCount_LastArrayItemKeptEmpty()
+        {
+            SqlMetaData[] metaData = new SqlMetaData[]
+            {
+                new SqlMetaData("col1", SqlDbType.NVarChar,50),
```
Comment (saurabh500) [inline src/System.Data.SqlClient/tests/FunctionalTests/SqlDataRecordTest.cs]: Nit: Space after `,` in `, 50`

**Comment by RemiBou** [comment]
Ok. I'll do that this week ( I thought I already did but I'll double check)

**Comment by RemiBou** [comment]
Visual Studio community didn't change anything after formating soI changed it by hand, now it should be alright

**Comment by saurabh500** [comment]
@dotnet-bot Test Linux x64 Release Build please 
Test OSX x64 Debug Build  please 
Test Tizen armel Debug Build please

**Comment by saurabh500** [comment]
Thanks @RemiBou 
Good to merge after CI is green.

**Comment by saurabh500** [comment]
@dotnet-bot test OSX x64 Debug Build please 

```
java.nio.channels.ClosedChannelException
Also:   hudson.remoting.Channel$CallSiteStackTrace: Remote call to JNLP4-connect connection from auth.prep.xbox.com/131.107.58.5:53307
		at hudson.remoting.Channel.attachCallSiteStackTrace(Channel.java:1693)
		at hudson.remoting.Request.call(Request.java:192)
		at hudson.remoting.Channel.call(Channel.java:907)
		at hudson.FilePath.act(FilePath.java:986)
		at hudson.FilePath.act(FilePath.java:975)
		at hudson.FilePath.chmod(FilePath.java:1591)
		at org.jenkinsci.plugins.durabletask.BourneShellScript.launchWithCookie(BourneShellScript.java:101)
		at org.jenkinsci.plugins.durabletask.FileMonitoringTask.launch(FileMonitoringTask.java:64)
		at org.jenkinsci.plugins.workflow.steps.durable_task.DurableTaskStep$Execution.start(DurableTaskStep.java:167)
		at org.jenkinsci.plugins.workflow.cps.DSL.invokeStep(DSL.java:224)
		at org.jenkinsci.plugins.workflow.cps.DSL.invokeMethod(DSL.java:150)
		at org.jenkinsci.plugins.workflow.cps.CpsScript.invokeMethod(CpsScript.java:108)
		at org.codehaus.groovy.runtime.callsite.PogoMetaClassSite.call(PogoMetaClassSite.java:48)
		at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:48)
		at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:113)
		at com.cloudbees.groovy.cps.sandbox.DefaultInvoker.methodCall(DefaultInvoker.java:19)
		at com.cloudbees.groovy.cps.impl.ContinuationGroup.methodCall(ContinuationGroup.java:57)
		at com.cloudbees.groovy.cps.impl.FunctionCallBlock$ContinuationImpl.dispatchOrArg(FunctionCallBlock.java:109)
		at com.cloudbees.groovy.cps.impl.FunctionCallBlock$ContinuationImpl.fixArg(FunctionCallBlock.java:82)
		at sun.reflect.GeneratedMethodAccessor162.invoke(Unknown Source)
		at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
		at java.lang.reflect.Method.invoke(Method.java:498)
		at com.cloudbees.groovy.cps.impl.ContinuationPtr$ContinuationImpl.receive(ContinuationPtr.java:72)
		at com.cloudbees.groovy.cps.impl.FunctionCallBlock$ContinuationImpl.dispatchOrArg(FunctionCallBlock.java:103)
		at com.cloudbees.groovy.cps.impl.FunctionCallBlock$ContinuationImpl.fixArg(FunctionCallBlock.java:82)
		at sun.reflect.GeneratedMethodAccessor162.invoke(Unknown Source)
		at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
		at java.lang.reflect.Method.invoke(Method.java:498)
		at com.cloudbees.groovy.cps.impl.ContinuationPtr$ContinuationImpl.receive(ContinuationPtr.java:72)
		at com.cloudbees.groovy.cps.impl.ContinuationGroup.methodCall(ContinuationGroup.java:60)
		at com.cloudbees.groovy.cps.impl.FunctionCallBlock$ContinuationImpl.dispatchOrArg(FunctionCallBlock.java:109)
		at com.cloudbees.groovy.cps.impl.FunctionCallBlock$ContinuationImpl.fixArg(FunctionCallBlock.java:82)
		at sun.reflect.GeneratedMethodAccessor162.invoke(Unknown Source)
		at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
		at java.lang.reflect.Method.invoke(Method.java:498)
		at com.cloudbees.groovy.cps.impl.ContinuationPtr$ContinuationImpl.receive(ContinuationPtr.java:72)
		at com.cloudbees.groovy.cps.impl.ConstantBlock.eval(ConstantBlock.java:21)
		at com.cloudbees.groovy.cps.Next.step(Next.java:83)
		at com.cloudbees.groovy.cps.Continuable$1.call(Continuable.java:174)
		at com.cloudbees.groovy.cps.Continuable$1.call(Continuable.java:163)
		at org.codehaus.groovy.runtime.GroovyCategorySupport$ThreadCategoryInfo.use(GroovyCategorySupport.java:122)
		at org.codehaus.groovy.runtime.GroovyCategorySupport.use(GroovyCategorySupport.java:261)
		at com.cloudbees.groovy.cps.Continuable.run0(Continuable.java:163)
		at org.jenkinsci.plugins.workflow.cps.SandboxContinuable.access$001(SandboxContinuable.java:19)
		at org.jenkinsci.plugins.workflow.cps.SandboxContinuable$1.call(SandboxContinuable.java:35)
		at org.jenkinsci.plugins.workflow.cps.SandboxContinuable$1.call(SandboxContinuable.java:32)
		at org.jenkinsci.plugins.scriptsecurity.sandbox.groovy.GroovySandbox.runInSandbox(GroovySandbox.java:108)
		at org.jenkinsci.plugins.workflow.cps.SandboxContinuable.run0(SandboxContinuable.java:32)
		at org.jenkinsci.plugins.workflow.cps.CpsThread.runNextChunk(CpsThread.java:174)
		at org.jenkinsci.plugins.workflow.cps.CpsThreadGroup.run(CpsThreadGroup.java:330)
		at org.jenkinsci.plugins.workflow.cps.CpsThreadGroup.access$100(CpsThreadGroup.java:82)
		at org.jenkinsci.plugins.workflow.cps.CpsThreadGroup$2.call(CpsThreadGroup.java:242)
		at org.jenkinsci.plugins.workflow.cps.CpsThreadGroup$2.call(CpsThreadGroup.java:230)
		at org.jenkinsci.plugins.workflow.cps.CpsVmExecutorService$2.call(CpsVmExecutorService.java:64)
		at java.util.concurrent.FutureTask.run(FutureTask.java:266)
		at hudson.remoting.SingleLaneExecutorService$1.run(SingleLaneExecutorService.java:112)
		at jenkins.util.ContextResettingExecutorService$1.run(ContextResettingExecutorService.java:28)
		at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
		at java.util.concurrent.FutureTask.run(FutureTask.java:266)
```

**Comment by stephentoub** [comment]
The macOS builds are currently broken in general. @dotnet/dnceng, where are we in getting them fixed?

**Comment by ahsonkhan** [comment]
Here is the issue tracking it - https://github.com/dotnet/core-eng/issues/2808

**Comment by saurabh500** [comment]
I will merge this PR because it doesn't test anything that is platform specific.

**Pull request merged**